api/api/src/main/scala/uk/ac/wellcome/platform/api/services/ImagesService.scala

```diff
@@ -50,10 +55,28 @@ class ImagesService(searchService: ElasticsearchService)(
     searchService
       .executeSearch(
         queryOptions = toElasticsearchQueryOptions(searchOptions),
+        requestBuilder = ImagesRequestBuilder,
         index = index
       )
       .map { _.map(createResultList) }
 
+  def retrieveSimilarImages(
+    index: Index,
+    image: AugmentedImage): Future[List[AugmentedImage]] =
+    searchService
+      .executeSearchRequest(
+        ImagesRequestBuilder.requestVisuallySimilar(
+          index = index,
+          id = image.id.canonicalId,
+          n = nVisuallySimilarImages
+        )
+      )
+      .map {
+        _.map {
+          _.hits.hits.map(hit => jsonTo(hit.sourceAsString)).toList
```

Comment:
I think with these nested anonymous functions it would be clearer if we defined the input arguments (i.e. `resp => ..` etc) rather than using the `_` syntax, as right now slightly hard to know what they refer to